auth/map.go

```diff
@@ -0,0 +1,22 @@
+package auth
+
+var (
+	PermissionStringMap map[Permission]string
+	PermissionEnumMap map[string]Permission
```

Comment:
no need to declare both variables here. you can pretty much have the following :
```golang
var (
	PermissionStringMap = map[Permission]string {
		NoPermission: "NO_PERM",
		UserRegistration: "USER_REGISTRATION",
		User: "USER",
		Admin: "ADMIN",
	}

 	PermissionEnumMap = map[string]Permission {
		PermissionStringMap[NoPermission]: NoPermission,
		PermissionStringMap[UserRegistration]: UserRegistration,
		PermissionStringMap[User]: User,
		PermissionStringMap[Admin]: Admin,
	}
)
```
This will declare and initialize.